src/scalar/union.rs

```diff
@@ -0,0 +1,48 @@
+use std::sync::Arc;
+
+use crate::datatypes::DataType;
+
+use super::Scalar;
+
+/// A single entry of a [`crate::array::UnionArray`].
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnionScalar {
+    value: Option<Arc<dyn Scalar>>,
+    type_: i8,
+    data_type: DataType,
+}
+
+impl UnionScalar {
+    /// Returns a new [`UnionScalar`]
+    #[inline]
+    pub fn new(data_type: DataType, type_: i8, value: Option<Arc<dyn Scalar>>) -> Self {
+        Self {
+            value,
+            type_,
+            data_type,
+        }
+    }
+
+    /// Returns the values irrespectively of the validity.
+    #[inline]
+    pub fn value(&self) -> &Option<Arc<dyn Scalar>> {
+        &self.value
+    }
+}
+
+impl Scalar for UnionScalar {
+    #[inline]
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    #[inline]
+    fn is_valid(&self) -> bool {
+        self.value.is_some()
```

Comment:
```suggestion
        true
```

union does not have validity, so it is always valid

Reply:
I think I misinterpreted the format spec. I thought unions did have validity based on the example for dense unions where one of the values is null (https://arrow.apache.org/docs/format/Columnar.html#union-layout) just not an explicit validity mask. I had initially used this in new_scalar to follow a similar pattern to the struct, but ended up going with a different implementation to avoid extra lookups.